client/login/wp-login/login-links.jsx

```diff
@@ -293,14 +294,10 @@ export class LoginLinks extends React.Component {
 			usernameOrEmail,
 		} = this.props;
 
-		const signupUrl = getSignupUrl(
-			query,
-			currentRoute,
-			oauth2Client,
-			locale,
-			pathname,
-			isGutenboarding
-		);
+		// use '?signup_url' if explicitly passed as URL query param
+		const signupUrl = this.props.signupUrl
+			? `${ window.location.origin }/${ this.props.signupUrl.replace( /^\//, '' ) }`
+			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname, isGutenboarding );
```

Comment:
I wonder 🤔 , isn't it possible to pass `${ window.location.origin }/${ this.props.signupUrl.replace( /^\//, '' ) }` as `pathname` argument to `getSignupUrl()`? If so, we can get rid of the ternary and call `getSignupUrl` directly.

Reply:
I don't really see the ternary here as offending. It makes sense for the consumer to know whether to call a function or not. Passing in a value to have it returned untouched may introduce complexity - if the function does anything with the URL before returning it (for this case at least) it would cause breakage. So not sure what ramifications may surface by going down that path.

Reply:
p.s. looks like we can remove the origin from the URL above. Just a relative path should suffice I think. 👀

Reply:
> I don't really see the ternary here as offending. It makes sense for the consumer to know whether to call a function or not.

I'm concerned about keeping the code DRY since the logic is already quite complicated, but calling the function just to pass and return the pathname is a bit too much. So let's keep the ternary as-is!

> p.s. looks like we can remove the origin from the URL above. Just a relative path should suffice I think. 👀

That sounds like a great idea. I'm curious to see how it works out 💯

Reply:
> I'm concerned about keeping the code DRY since the logic is already quite complicated

Not every condition needs to be encapsulated, and as a general principle, we should be mindful about creating complexity from abstractions. Passing a signup URL to `getSignupUrl` to have it returned immediately feels very much along those lines. 🤷‍♂️

Reply:
> That sounds like a great idea. I'm curious to see how it works out 💯

Should be addressed in the last commit. It works out 😁

Reply:
> Should be addressed in the last commit. It works out 😁

Nope. Puts us back into vulnerable territory. at least not without additional checks. reverting.

Reply:
I will see about opening a security report/issue for any other `signupUrl` uses that result in redirections.